Api/ResourceResolverTrait.php

```diff
@@ -54,7 +54,9 @@ public function guessResource($object, array $context = null, $strict = false)
             $type = gettype($type);
         }
 
-        if (isset($context['resource'])) {
+        if (is_array($object) && isset($object['@type'])) {
```

Comment:
It cannot work because `@type` can be any IRI like `http://schema.org/Product`. And it's why tests fail.